decontamination-authentication.qmd

```diff
@@ -0,0 +1,359 @@
+
+If one wants to assess the degree of contamination for each sample, there is a handy tool [cuperdec](https://github.com/jfy133/cuperdec), which is an R package that allows a quick comparison of microbial profiles in a query metagenomic sample against a database. The idea of *cuperdec* is to rank organisms in each sample by their abundance and then using an "expanding window" approach to compute their enrichment in a reference database that contains a comprehensive list of microbial organisms which are specific to a tissue / environment in question. The tool produces so-called *Cumulative Percent Decay* curves that aim to represent the level of endogenous content of microbiome samples, such as ancient dental calculus, to help to identify samples with low levels of preservation that should be discarded for downstream analysis.
+
+```R
+library("cuperdec")
+library("magrittr")
+library("dplyr")
+
+# Load database (in this case oral database)
+data(cuperdec_database_ex)
+database <- load_database(cuperdec_database_ex, target = "oral") %>% print()
+
+# Load abundance matrix and metadata
+taxatable <- load_taxa_table("abund_matrix.txt")  %>% print()
+metadata <- as_tibble(data.frame(Sample = unique(taxatable$Sample), 
+Sample_Source = "Oral"))
+
+# Compute cumulative percent decay curves, filter and plot results
+curves <- calculate_curve(taxatable, database = database) %>% print()
+filter_result <- simple_filter(curves, percent_threshold = 50) %>% print()
+plot_cuperdec(curves, metadata = my_metadata_map, filter_result)
+```
+
+![](assets/images/chapters/decontamination-authentication/cuperdec1.png)
+
+In the figure above, one curve represents one sample, and the red curves have a very high amount of contamination and very low amount of endogenous DNA. These samples might be considered to be dropped from the downstream analysis.
+
+## Source tracking
+
+For the case of ancient microbiome profiling, in addition to traditional inspection of the list of detected organisms and comparing it with the ones detected on blanks, it is useful to use pre-trained classifiers that can give a prediction on what environment the detected organisms most likely come from. The most popular and widely used tool is called [**SourceTracker**](https://www.nature.com/articles/nmeth.1650#citeas). SourceTracker is a Bayesian version of the Gaussian Mixture Model (GMM) clustering algorithm that is trained on a reference data called **Sources**, i.e. different classes such as Soil or Human Oral or Human Gut microbial communities etc., and then it can estimate proportion / contribution of each Source into test samples called **Sinks**.
```

Comment:
```suggestion
For the case of ancient microbiome profiling, in addition to traditional inspection of the list of detected organisms and comparing it with the ones detected on blanks, we can use tools that make a prediction on what environment the detected organisms most likely come from. 

The most popular and widely used tool is called [**SourceTracker**](https://www.nature.com/articles/nmeth.1650#citeas). SourceTracker is a Bayesian version of the Gaussian Mixture Model (GMM) clustering algorithm that is trained on a user-supplied reference data called **Sources**, i.e. different classes such as Soil or Human Oral or Human Gut microbial communities etc., and then it can estimate proportion / contribution of each of these sources the users actual samples called **Sinks**.
```

Reply:
I just noted that throughout the decontamination section you're not really mentioning what type of input data are required (just imlpied...) e.g. for decontam requires taxonomic profiles and a metadta sheet, cuperdec also requries taxon tables etc., - a bit like how you've done in sourcetracker below!)